src/MOI_wrapper.jl

```diff
@@ -2488,3 +2488,65 @@ function MOI.get(
     end
     return get_intattrelement(model.inner, "IISQConstr", _info(model, index).row) > 0
 end
+
+###
+### Constraint attributes
+###
+
+# Gurobi constraint attributes as documented at https://www.gurobi.com/documentation/8.1/refman/linear_constraint_attribut.html
+const CONSTR_ATTR_TYPE =
+    Dict( "Lazy" => Integer, "DStart" => Real, "IISConstr" => Integer,
+          "SARHSLow" => Real, "SARHSUp" => Real, "FarkasDual" => Real,
+          "Sense" => Char, "RHS" => Real, "ConstrName" => String, "Pi" => Real,
+          "Slack" => Real, " CBasis" => Integer )
+const GETTER_FOR_TYPE = Dict( Integer => Gurobi.get_intattrelement,
+                              Real => Gurobi.get_dblattrelement,
+                              Char => Gurobi.get_charattrelement,
+                              String => Gurobi.get_strattrelement)
+const SETTER_FOR_TYPE = Dict( Integer => Gurobi.set_intattrelement!,
+                              Real => Gurobi.set_dblattrelement!,
+                              Char => Gurobi.set_charattrelement!,
+                              String => Gurobi.set_strattrelement! )
+
+struct ConstraintAttribute <: MOI.AbstractConstraintAttribute
+    name::String
+end
+
+MOI.supports(::Optimizer, attr::ConstraintAttribute, ::Type{<:MOI.ConstraintIndex}) = attr.name ∈ keys(CONSTR_ATTR_TYPE)
+
+"""
+MOI.set(model::Optimizer, attr::ConstraintAttribute,
+        ci::MOI.ConstraintIndex{MOI.ScalarAffineFunction{Float64}, <:Any},
+        value::T) where T
+
+Set a constraint attribute.
+    
+Checks that the attribute exists and that value is correctly typed, but lets
+Gurobi handle cases where an attribute's value cannot be set.
+
+Caveat: might fail due to incorrect type for an attribute that cannot be set
+anyway.
+"""
+function MOI.set(
+    model::Optimizer, attr::ConstraintAttribute,
+    ci::MOI.ConstraintIndex{MOI.ScalarAffineFunction{Float64}},
+    value::T
+) where T
+    MOI.supports(model, attr, typeof(ci)) ||
+        throw(MOI.UnsupportedAttribute(attr))
+    T <: CONSTR_ATTR_TYPE[attr.name] || throw(ArgumentError("Attribute $(attr.name) is $(CONSTR_ATTR_TYPE[attr.name]) but $T provided."))
+    setter! = SETTER_FOR_TYPE[CONSTR_ATTR_TYPE[attr.name]]
+    setter!(model.inner, attr.name, _info(model, ci).row, value)
+    _require_update(model) 
+    return
+end
+
+function MOI.get(
+    model::Optimizer, attr::ConstraintAttribute,
+    ci::MOI.ConstraintIndex{MOI.ScalarAffineFunction{Float64}}
+)
+    attr.name ∈ keys(CONSTR_ATTR_TYPE) || throw(MOI.UnsupportedAttribute(attr))
```

Comment:
This should also be replaced by `supports`

Reply:
Now done in https://github.com/JuliaOpt/Gurobi.jl/pull/243/commits/27cf7893e3bf15be23bbaf0d31c1dd78ce6bafcd - am super distracted it seems